app/controllers/search_controller.rb

```diff
@@ -7,41 +8,83 @@ def index
 
     @trips ||= []
     @search = Search.new(params[:search])
+    the_number_of_trips_per_page = 10
+    the_first_date = [(@search.date_value) -31, (Date.today) -31].min # we need trips of the past month to say "hey there was some trips last month" if no results
+    the_last_date =  (@search.date_value) +31
```

Comment:
Parenthesis unnecessary

Reply:
Are you sure ? .date_value is a function, ```@search.date_value -31``` crashes for me.